internal/backend/service/user.go

```diff
@@ -93,40 +69,45 @@ func (s *UserServer) ListUserHistory(
 		slogger = slog.Default()
 	}
 
-	groupID, err := interceptor.GetGroupID(ctx)
-	if err != nil {
-		slogger.Error("Couldn't get group id", "Error", err)
-		return nil, connect.NewError(connect.CodeInternal, err)
+	// Get server ID from header
+	serverID := req.Header().Get(ServerIDHeader)
+	if serverID == "" {
+		return nil, connect.NewError(connect.CodeInvalidArgument, 
+			fmt.Errorf("server ID header is required"))
 	}
 
-	slogger.Info("Group ID", "ID", groupID)
-
-	db, err := group.GetGroupDB(ctx, s.tokenCache.Get(), s.libSQLConfig, groupID)
+	// Find group ID for this server
+	groupID, err := s.dbClient.FindGroupByServer(ctx, serverID)
 	if err != nil {
-		slogger.Error("Failed getting group db", "Error", err)
-		return nil, connect.NewError(connect.CodeInternal, err)
+		slogger.Error("Failed to find group for server", "server_id", serverID, "error", err)
+		return nil, connect.NewError(connect.CodeNotFound, err)
 	}
-	queries := groupSQLc.New(db)
-
-	dbUserHistory, err := queries.GetUserHistory(ctx, req.Msg.UserId)
 
+	// Get user history
+	dbEntries, err := s.dbClient.GetUserHistory(ctx, groupID, req.Msg.UserId, nil, nil)
 	if err != nil {
-		slogger.Error(fmt.Sprintf("failed to get user history %s", req.Msg.UserId), "Error", err)
+		slogger.Error("Failed to get user history", "group_id", groupID, "user_id", req.Msg.UserId, "error", err)
 		return nil, connect.NewError(connect.CodeInternal, err)
 	}
 
-	rpcHistory := make([]*snitchv1.CreateUserHistoryRequest, 0, len(dbUserHistory))
-
-	for _, userHistory := range dbUserHistory {
-		rpcHistory = append(rpcHistory, &snitchv1.CreateUserHistoryRequest{
-			Username:   userHistory.Username,
-			ChangedAt:  userHistory.ChangedAt,
-			UserId:     userHistory.UserID,
-			GlobalName: userHistory.GlobalName,
+	// Convert to old API format (temporarily)
```

Comment:
What does this mean by old API format? I saw a similar comment once previously too